src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,69 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_VALUE = 100;
+    public static final int DEFAULT_RADIUS = 10;
+    public static final Color DEFAULT_COLOR = Color.WHITE;
+
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private Models models;
+
+    public Figure getRandomFigure() {
+        models = Models.values()[random.nextInt(Models.values().length)];
+        switch (models) {
+            case CIRCLE:
+                return getRandomCircle();
+            case ISOSCELES_TRAPEZOID:
+                return getRandomIsoscelesTrapezoid();
+            case RECTANGLE:
+                return getRandomRectangle();
+            case RIGHT_TRIANGLE:
+                return getRandomRightTriangle();
+            case SQUARE:
+                return getRandomSquare();
+            default:
+                return getRandomFigure();
```

Comment:
method getRandomFigure shouldn't return default figure, if you checked all figures except square then we know that square should be returned
```suggestion
            default:
                return getRandomSquare();
```